core/domain/classifier_services.py

```diff
@@ -127,6 +127,52 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        create_classifier_training_job(algorithm_id, interaction_id,
+                                       exp_id, exp_version, state_name,
+                                       training_data,
+                                       feconf.TRAINING_JOB_STATUS_NEW)
+
+
+def update_classifier_exploration_mappings(exploration, state_names):
+    """Creates new ClassifierExplorationMappingModel instances for all the
+    state names passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
```

Comment:
Let me check my understanding. What you want to do is to map the classifier for the newly-created version to the same one that was used for the previous version, so that you don't need to regenerate a new classifier, right?

If so, then I think just passing in the exp_id, the current version, and the list of state_names should suffice. Check if the classifier for `current_version - 1` exists. If so, get its classifier ID and use that. If not, some sort of error should happen, I guess -- or maybe just delegate to the function that creates classifiers from scratch instead?

One issue to consider here is the following: suppose someone saves exp v1, and then makes a trivial edit and saves exp v2 before the classifiers themselves have been created. Will the classifier ID be known by then? If not, what do we want to do in that case?